### PR TITLE
#28 Proposed changes of adding Record Key in `list` using `Formatter`

### LINKED ISSUE #28: `timetrace list records` should include "record key"
The table displayed by `timetrace list records` should contain a new column "key". This should be the "record key" in the form `YYYY-MM-DD-HH-MM` - i.e. the exact value that you would pass to `timetrace get record` to display that record.

### PR DESCRIPTION
**Pull request by aaronsheah**
### 12 Hour
Screenshot 2021-05-19 at 23 40 32

### 24 Hour
Screenshot 2021-05-19 at 23 42 11

resolves #28

### CODE REVIEW

File: core/formatter.go [hunk 1]
```diff
@@ -21,3 +21,23 @@ func (f *Formatter) timeLayout() string {
 func (f *Formatter) TimeString(input time.Time) string {
 	return input.Format(f.timeLayout())
 }
+
+const (
+	defaultRecordKeyLayout        = "2006-01-02-15-04"
+	default12HoursRecordKeyLayout = "03:04PM"
```
Comment (dominikbraun) [inline core/formatter.go:27]: I think this record key layout should be `2006-01-02-03-04PM`, shouldn't it?

File: core/formatter.go [hunk 2]
```diff
@@ -21,3 +21,23 @@ func (f *Formatter) timeLayout() string {
...
+)
+
+func (f *Formatter) recordKeyLayout() string {
+	if f.use12Hours {
+		return defaultRecordKeyLayout
+	}
+	return default12HoursRecordKeyLayout
```
Comment (dominikbraun) [inline core/formatter.go:34]:
```suggestion
	if f.use12Hours {
		return default12HoursRecordKeyLayout
	}
	return defaultRecordKeyLayout
```

**Review by dominikbraun — CHANGES_REQUESTED** [review]
@aligator PTAL

**Comment by aaronsheah** [comment]
Thanks @dominikbraun silly mistakes, must have still haven't woken up then 😆

**Comment by dominikbraun** [comment]
@aaronsheah I think you need to use the new Formatter for `get` and `delete` as well - then this PR can be merged 😄

**Comment by aligator** [comment]
> @aligator PTAL

Looks good to me :-)

**Comment by aaronsheah** [comment]
@dominikbraun updated :)

**Review by dominikbraun — APPROVED** [review]
LGTM 👍

**Pull request merged**